jcl/src/openj9.jvm/share/classes/com/ibm/jvm/CRIUSupport.java

```diff
@@ -43,4 +77,49 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 	
+	/**
+	 * Checkpoint the JVM
+	 * 
+	 * @param cpDataDir the directory in which the checkpoint data will be stored
+	 * 
+	 * @return return CRIUOpRetCode
+	 */
+	public static CRIUOpRetCode checkPointJVM(String cpDataDir) {
+		CRIUOpRetCode ret = CRIUOpRetCode.SUCCESS;
+		
+		if (!isCRIUSupportEnabled()) {
+			ret = CRIUOpRetCode.UNSUPPORTED_OPERATION;
+		} else {
+			switch(checkpointJVMImpl(cpDataDir, false, true, true, 4, "checkpoint.log")) { //$NON-NLS-1$ //$NON-NLS-2$
+			case 1:
+				ret = CRIUOpRetCode.UNSUPPORTED_OPERATION;
+				break;
+			case 2:
+				ret = CRIUOpRetCode.INVALID_ARGUEMENTS;
+				break;
+			case 3:
+				ret = CRIUOpRetCode.SYSTEM_CHECKPOINT_FAILURE;
+				break;
+			case 4:
+				ret = CRIUOpRetCode.JVM_CHECKPOINT_FAILURE;
+				break;
+			case 5:
+				ret = CRIUOpRetCode.JVM_RESTORE_FAILURE;
+				break;
+			default:
+				/* success */
+			}
+		}
+		
+		return ret;
+	}
+	
+	/**
+	 * Checkpoint the JVM. Checkpoint data is stored in [currentDir]/cpData.
+	 * 
+	 * @return return CRIUOpRetCode
+	 */
+	public static CRIUOpRetCode checkPointJVM() {
+		return checkPointJVM(Paths.get("cpData").toAbsolutePath().toString());
```

Comment:
Writing to the current dir doesn't seem ideal, since this could be changed. What about making this relative to the "user.dir" instead? Also, this should have a permission check as well.

Reply:
I'd also avoid short directory names like `cpData` as they don't mean anything to users.  Easier for users as they look in the `ls` output and see `jvm_checkpoints` or `j9_checkpoints` (or similar) as the directory name